### DISCUSSION TITLE
Supabase Cli start - Decrease start up time in Github Action

### DISCUSSION THREAD

**Discussion by MJDeligan** [supabase/supabase#9351] [Questions]
Hey, I'm currently working on a project where I'm using Github Actions for CI/CD with Supabase. It's working well, but one of the jobs before deploying migrations to production is to do e2e testing using a supabase instance that is spun up in the action runner. I'm doing this by setting up the CLI using the provided action and then running `supabase start`. This is very slow though and takes the runtime for the action from 40-60 seconds up to 5 minutes.

I see that a lot of services are being started and their images probably being pulled. Some of them, like studio and postgrest, aren't required for the e2e-tests, but I see no way of disabling them. I'm not good with Github Actions and Docker so I'd like to know if there are strategies to speed up the `supabase start` command in a CI-context.

**Comment by aaronzhongg** [comment]
You can exclude starting unnecessary containers using the `--exclude` arg. I.e,
`supabase start --exclude postgrest,studio`

https://supabase.com/docs/reference/cli/supabase-start?example=supabase-start-without-studio

It reduces my startup time slightly, but still takes a few minutes. Would be great to be able to improve it more

**Comment by asjir** [comment]
I know it's not answering the question, but starting it concurrently works, i.e. like this:
```yaml
      - name: Start Supabase in background
        run: |
          supabase start --exclude realtime,storage-api,imgproxy,inbucket,studio,edge-runtime &
          echo "Supabase starting in background..."
... all the build steps here
      - name: Wait for Supabase to be ready
        run: |
          timeout=180
          counter=0
          until curl -s http://localhost:54321/rest/v1/ >/dev/null; do
            if [ $counter -gt $timeout ]; then
              echo "Timed out waiting for Supabase"
              exit 1
            fi
            echo "Waiting for Supabase to be ready..."
            sleep 1
            counter=$((counter + 1))
          done
          echo "Supabase is ready!"
```
and in my case it takes roughly the same time as typechecking and build steps of the app so I'm no longer blocked by it

**Reply by dohooo** [reply]
You are a genius!!!!!!!

**Reply by gkawin** [reply]
Love this!!! It saved running time almost hit 50%.

**Comment by neuromaxer** [comment]
I would bump this topic and ask for an official response if it's possible to somehow speed up the supabase client startup (docker image pulling) in CI besides removing some images and concurrent loading as currently it takes ~3 minutes which is quite long

**Comment by richikchanda1999** [comment]
It takes around 3 minutes now, while the tests take up just a minute in my case. Staring supabase is becoming the bottleneck.
